File: AddFraction.cpp
#include "stdafx.h"
#include "AddFraction.h"


AddFraction::AddFraction(vector<unsigned> num, vector<unsigned> denom) {
	numerators = num;
	denominators = denom;
	answer = 0;
}

void AddFraction::canonizeFraction() {
	// find their LCM
	LCM l_c_m(denominators);
	lcm = l_c_m.getLCM();

	// transform fractions so they all have same denominator
	for (unsigned i = 0; i < denominators.size(); i++) {
		new_numerators.push_back(lcm / denominators[i] * numerators[i]);
	}
	new_numerators.shrink_to_fit();
}

void AddFraction::doAdd() {
	canonizeFraction();

	// add all transformed numerators
	for (unsigned i = 0; i < new_numerators.size(); i++) {
		answer += new_numerators[i];
	}
}


AddFraction::~AddFraction() {
}

File: AddFraction.js
var numerators = [1, 1, 1, 1];//enter your own numerators
var denominators = [4, 4, 4, 4];//enter your own denominators
var new_numerators = [];
var answer = 0;
var LCM = 0;

all_factors = [];
state_check = false;
i = 2;
index = 0;

function canonizeFraction() {
    // find the LCM of denominators
    // STEP 1:
    LCM = getLCM(denominators);

    // transform fractions so they all have same denominator
    // STEP 2:
    for (var i = 0; i < denominators.length; i++) {
        new_numerators[i] = LCM / denominators[i] * numerators[i];
    }
}

function doAdd() {
    canonizeFraction();

    // add all transformed numerators
    // STEP 3:
    for (var i = 0; i < new_numerators.length; i++) {
        answer += new_numerators[i];
    }
}
